pkg/services/librarypanels/database.go

```diff
@@ -110,6 +131,26 @@ func (lps *LibraryPanelService) disconnectDashboard(c *models.ReqContext, uid st
 	})
 }
 
+// disconnectLibraryPanelsForDashboard deletes connections for all Library Panels in a Dashboard.
+func (lps *LibraryPanelService) disconnectLibraryPanelsForDashboard(dashboardID int64, panelCount int64) error {
+	if panelCount == 0 {
```

Comment:
This should be removed, since it's subject to race conditions, i.e. connections may have been added by a parallel process.

Reply:
Hmm, it's hard to tell what you want me to remove. Do you mean the if statement?

Reply:
@aknuds1 I think you need to clarify this because I don't really understand.

Reply:
I mean the `if` statement. It makes an assumption that may be wrong, i.e. we don't know the number of panels at this point in time.

Reply:
It's connected to my [other](https://github.com/grafana/grafana/pull/30278#discussion_r560729613) comment, i.e. both are subject to the same race condition (another process may have added or removed connections).

Reply:
Ok, I need to switch to multithreaded thinking here. Good catch!